backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisSessionManager.java

```diff
@@ -7,97 +7,78 @@
 import org.joda.time.DateTime;
 import org.joda.time.Minutes;
 import org.molgenis.emx2.Database;
-import org.molgenis.emx2.MolgenisException;
 import org.molgenis.emx2.sql.SqlDatabase;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import spark.Request;
-import spark.Response;
 
 public class MolgenisSessionManager {
   public static final String MOLGENIS_TOKEN = "x-molgenis-token";
-  private static final String SESSION_ATTRIBUTE = "session";
+  private static final String SESSION_ATTRIBUTE = "molgenis_session";
   private static final Logger logger = LoggerFactory.getLogger(MolgenisSessionManager.class);
+  private static final int SESSION_TIMEOUT = 30;
 
-  // key is the user, might lead to trouble
+  // map so we can track the sessions
+  // session id is the key, todo in case of session less requests outside browser
   private Map<String, MolgenisSession> sessions = new LinkedHashMap<>();
 
   public MolgenisSessionManager() {}
 
-  public MolgenisSession getSession(Request request) {
-    // already in a session, then return that
-    if (request.session().attribute(SESSION_ATTRIBUTE) != null) {
-      MolgenisSession session = request.session().attribute(SESSION_ATTRIBUTE);
-      // timeout
-      if (minutesBetween(session.getCreateTime(), DateTime.now())
-          .isGreaterThan(Minutes.minutes(30))) {
-        request.session(false); // destroy session
-        if (logger.isInfoEnabled()) {
-          logger.info(
-              "Destroyed session for user({0}) because timeout more than 30mins",
-              session.getSessionUser());
-        }
-
-      } else {
-        logger.info("Reusing session for user({})", session.getSessionUser());
-        // refresh timeout
-        session.setCreateTime(DateTime.now());
-        return session;
+  public synchronized MolgenisSession getSession(Request request) {
```

Comment:
This is the easiest way to make things threadsafe but means the session manager can only look up one session at a time, which hurts performance for concurrent requests

Reply:
I believe we dont really need to make it synchronized; instead we can use an optimistic approach where session object might be set more than once

Reply:
refactored to let jetty take care of session management